amulet/deployer.py

```diff
@@ -330,6 +337,53 @@ def deploy_w_timeout_and_dir(self, timeout, deploy_dir):
             yield
         self.deployed = True
 
+    def action_defined(self, service):
+        """
+        :param service: Name of service to get list of defined actions for.
+
+        Returns the list of actions defined for the service.
+        """
+        if service not in self.services:
+            raise ValueError('Service needs to be added before listing actions.')
+        raw = juju(['action', 'defined', service, '--format', 'json'])
+        return json.loads(raw)
+
+    def action_do(self, unit, action, **kw):
```

Comment:
I'd also argue that action should be less a Deployment method and more a Sentry command

```
id = d.sentry['unit/#'].do(action, config)
d.action_fetch(id)
```

but I'm not opposed to the current format (and may even just add a wrapper to UnitSentry later as a short cut)